### ISSUE TITLE
pause: seconds=0 waits for input instead of finishing immediately

### ISSUE THREAD

**Issue by mmaslowskicc** [ansible/ansible#16457]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

Default.
##### OS / ENVIRONMENT

Debian Jessie amd64
##### SUMMARY

When running the task `pause: seconds=0`, `ansible-playbook` hangs instead of immediately finishing the pause.

This results from using `signal.alarm` with an argument of `0` which cancels an alarm instead of setting it.
##### STEPS TO REPRODUCE

Playbook `play.yml`:

```

---
- hosts: localhost
  gather_facts: false
  tasks:
     - pause: seconds=0
```

(A real example would get the `0` from a variable which sometimes has zero, sometimes a larger value.)

Run

```
ansible-playbook -i /dev/null play.yml
```

##### EXPECTED RESULTS

When running Ansible 1.9.4:

```
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ************************************************************** 

TASK: [pause seconds=0] ******************************************************* 
(^C-c = continue early, ^C-a = abort)
[localhost]
Pausing for 0 seconds
ok: [localhost]

PLAY RECAP ******************************************************************** 
localhost                  : ok=1    changed=0    unreachable=0    failed=0
```

The pause ended without needing any interaction.
##### ACTUAL RESULTS

```
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************

TASK [pause] *******************************************************************
Pausing for 0 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
```

Here it just waits for `ctrl+C`. After pressing it and `C`:

```
Press 'C' to continue the play or 'A' to abort 
ok: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0
```

**Comment by bcoca** [comment]
Module Repo Information
=======================

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a submission about a module, and aside from action_plugins, if you know what those are, the modules
in ansible are now moved to separate repos.  We would appreciate if you can submit this there instead.

If this is about a new module, submit pull requests or ideas to:

   * https://github.com/ansible/ansible-modules-extras

If this is about an existing module:

   * Find the module at http://docs.ansible.com/list_of_all_modules.html
   * Open the documentation page for that module
   * If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
   * Otherwise, submit your module to https://github.com/ansible/ansible-modules-core

Additionally, should you need more help with this, you can ask questions on:

   * IRC: #ansible on irc.freenode.net   
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

Thanks!

**Comment by mmaslowskicc** [comment]
`pause` is an action plugin, so I assumed it should be reported here.

**Comment by bcoca** [comment]
fixed via 1b7b1ca15e4a578c5ef788ae2e1d1d13b2d61a32, also updated the modules-core repo with docs explaining the behavior.
